demos/source/aws_iot_demo_jobs.c

```diff
@@ -0,0 +1,533 @@
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file aws_iot_demo_jobs.c
+ * @brief Demonstrates use of the AWS Jobs library.
+ */
+
+/* Config include. Should always come first per the style guide. */
+#include "iot_config.h"
+
+/* System Includes */
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/* Demo logging. */
+#include "iot_demo_logging.h"
+
+/* Platform Includes. */
+#include "platform/iot_clock.h"
+#include "platform/iot_network.h"
+#include "platform/iot_threads.h"
+
+/* MQTT Include. The underlying protocol for this demo. */
+#include "iot_mqtt.h"
+
+/* JSON Utilies for pulling out values. */
+#include "iot_json_utils.h"
+
+/* Jobs Library. */
+#include "aws_iot_jobs.h"
+#include "types/aws_iot_jobs_types.h"
+
+/**
+ * @brief The Jobs Library MQTT timeout.
+ * @TODO: Figure out how this interacts with the MQTT library timeout.
+ */
+#define AWS_IOT_DEMO_JOBS_MQTT_TIMEOUT ( 1000U )
+
+/**
+ * @brief The MQTT Library timeout.
+ */
+#define TIMEOUT_MS ( 5000u )
+
+/**
+ * @brief The MQTT Library keepalive time.
+ */
+#define KEEP_ALIVE_SECONDS ( 60 )
+
+/**
+ * @brief The key of the Job ID.
+ */
+#define JOB_ID_KEY        "jobId"
+#define JOB_ID_KEY_LENGTH ( sizeof( JOB_ID_KEY ) - 1 )
+
+/**
+ * @brief Keys for the Job JSON response.
+ */
+#define JOB_DOC_KEY                  "jobDocument"
+#define JOB_DOC_KEY_LENGTH           ( sizeof( JOB_DOC_KEY ) - 1 )
+
+#define JOB_ACTION_KEY               "action"
+#define JOB_ACTION_KEY_LENGTH        ( sizeof( JOB_ACTION_KEY ) - 1 )
+
+#define JOB_MESSAGE_KEY              "message"
+#define JOB_MESSAGE_KEY_LENGTH       ( sizeof( JOB_MESSAGE_KEY ) - 1 )
+
+/**
+ * @brief Max Lengths for JSON values.
+ */
+#define JOBS_DEMO_MAX_ID_LENGTH      64
+#define JOBS_DEMO_MAX_JOB_DOC_LENGTH 64
+/**
+ * @brief The previous JobId and length
+ */
+static char _pJobId[ JOBS_DEMO_MAX_ID_LENGTH + 1 ] = {0};
+static size_t _JobIdLength = 0;
+
+/**
+ * @brief The Job Document which will be passed into the Notify Next callback.
+ */
+static char _jobDoc[ JOBS_DEMO_MAX_JOB_DOC_LENGTH + 1 ] = {0};
+static size_t _jobDocLength = 0;
+
+/**
+ * @brief Execute the Jobs Demo.
+ *
+ * This is the top level function called by the IOT Demo harness.
+ */
+int32_t RunJobsDemo( bool awsIotMqttMode,
```

Comment:
Return value should just be `int`, for compatibility with `EXIT_SUCCESS` and `EXIT_FAILURE`